### PR TITLE
Ship packages as zip instead of tar.gz

### LINKED ISSUE #474: Discuss: Move to zip instead of `tar.gz` for the format of the package.
Currently, the package registry uses the `tar.gz` as the compressed package format for shipping the packages down to Kibana. Sadly this format is not supported directly by Windows. It would be nice if we would move to a zip file format this would allow anyone to download the package on windows unzip it and inspect the content.

@ruflin I vaguely remember you mentioning that there were some issues in Kibana that prevent us from moving to ZIP.

### PR DESCRIPTION
**Pull request by ruflin**
This is a breaking change and first needs changes on the Kibana side. The new registry and the new version of Kibana should be either rolled out in sync or Kibana supports both for a bit.

Closes #474

### CODE REVIEW

**Comment by ruflin** [comment]
Opening this as ready for review even though CI will be red on the integration tests. The reason is that Kibana does not support zip at the moment.

**Comment by mtojek** [comment]
Do you think it's feasible to prepare an archiver abstraction and support two modes (zip, tar.gz)?

**Comment by ruflin** [comment]
@mtojek Definitively possible but what would be the advantage of having the abstraction if we only plan to support one?

**Comment by mtojek** [comment]
Last time we also planned to support single one :) In the description you've written that we can't push it until Kibana supports the zip format. With an appropriate abstraction we don't need to wait for this.

**Comment by ruflin** [comment]
I realised, only Kibana can handle both as it is a single entry in the registry output (`download`). Lets not add abstraction for it before we are sure we will need it in the future. 

As soon as https://github.com/elastic/kibana/pull/76197 is merged I'll trigger CI again to see if it goes green.

**Comment by mtojek** [comment]
The implementation looks good to me, let's wait for the CI to pass all tests.

**Comment by ruflin** [comment]
jenkins, test this

**Comment by ruflin** [comment]
With 7.10 docker images from Kibana updated, this now seems to pass CI too and should be ready to be merged. @jfsiii @mtojek Would be good to get from both of you the 👍

File: archiver/archive.go [hunk 1]
```diff
@@ -87,24 +80,20 @@ func ArchivePackage(w io.Writer, properties PackageProperties) (err error) {
...
 
-	err = tarWriter.Flush()
-	if err != nil {
-		return errors.Wrap(err, "flushing tar writer failed")
-	}
-
-	err = gzipWriter.Flush()
+	err = zipWriter.Flush()
 	if err != nil {
 		return errors.Wrap(err, "flushing gzip writer failed")
```
Comment (mtojek) [inline archiver/archive.go:85]: nit: zip writer

**Review by mtojek — APPROVED** [review]
single nit-pick only

File: archiver/archive.go (hunk 1 above)
Comment (ruflin) [reply archiver/archive.go:85]: fixed.

**Pull request merged**